app/controllers/reliefs/ChooseReliefsController.scala

```diff
@@ -16,31 +16,30 @@
 
 package controllers.reliefs
 
-import config.FrontendDelegationConnector
 import connectors.{BackLinkCacheConnector, DataCacheConnector}
 import controllers.BackLinkController
-import controllers.auth.{AtedFrontendAuthHelpers, AtedRegime, ClientHelper}
+import controllers.auth.{AuthAction, ClientHelper}
 import forms.ReliefForms._
 import models.Reliefs
 import play.api.Play.current
 import play.api.i18n.Messages.Implicits._
-import services.ReliefsService
-import uk.gov.hmrc.play.frontend.auth.DelegationAwareActions
-import utils.PeriodUtils
+import play.api.mvc.{Action, AnyContent}
+import services.{DelegationService, ReliefsService}
 import utils.AtedUtils._
+import utils.PeriodUtils
 
 import scala.collection.mutable.ArrayBuffer
 
 trait ChooseReliefsController extends BackLinkController
-  with AtedFrontendAuthHelpers with ReliefHelpers with DelegationAwareActions  with ClientHelper {
+    with ReliefHelpers with ClientHelper with AuthAction {
 
-  def view(periodKey: Int) = AuthAction(AtedRegime) {
-    implicit atedContext =>
+  def view(periodKey: Int): Action[AnyContent] = Action.async { implicit request =>
+    authorisedAction { implicit authContext =>
       ensureClientContext {
         validatePeriodKey(periodKey) {
           for {
             backLink <- currentBackLink
-            retrievedData <- reliefsService.retrieveDraftReliefs(atedContext.user.atedReferenceNumber, periodKey)
+            retrievedData <- reliefsService.retrieveDraftReliefs(authContext.atedReferenceNumber.get, periodKey)
```

Comment:
RED - None.get risk. 

To stop repetition, just fix all instances of it, I won't point out anymore.

Reply:
Removed and refactored to use String as opposed to Option String